src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -760,13 +770,50 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             #using-trait-objects-that-allow-for-values-of-different-types>";
         let has_dyn = snippet.split_whitespace().next().map_or(false, |s| s == "dyn");
         let trait_obj = if has_dyn { &snippet[4..] } else { &snippet[..] };
-        if all_returns_have_same_type {
+        if count == 0 {
+            // No return paths. Suggest `-> T`, `-> impl Trait`, and if `Trait` is object safe,
+            // `-> Box<dyn Trait>`.
+            err.note(
+                "currently nothing is being returned, depending on the final implementation \
+                 you could change the return type in different ways",
+            );
+            err.span_suggestion(
+                ret_ty.span,
+                "you could use some type `T` that is `T: Sized` as the return type if all return \
```

Comment:
```suggestion
                "use some type `T` that is `T: Sized` as the return type if all return \
```

Reply:
The reason I kept that wording was to make it clear that the different suggestions are alternatives. That being said, the later "... if ..." already does that.